readthedocs/search/api/v3/views.py

```diff
@@ -0,0 +1,134 @@
+from functools import cached_property
+
+import structlog
+from django.utils import timezone
+from django.utils.translation import gettext as _
+from rest_framework.exceptions import ValidationError
+from rest_framework.generics import GenericAPIView
+
+from readthedocs.core.utils.extend import SettingsOverrideObject
+from readthedocs.search import tasks
+from readthedocs.search.api.pagination import SearchPagination
+from readthedocs.search.api.v3.backend import Backend
+from readthedocs.search.api.v3.serializers import PageSearchSerializer
+from readthedocs.search.api.v3.utils import should_use_advanced_query
+
+log = structlog.get_logger(__name__)
+
+
+class SearchAPI(GenericAPIView):
+
+    """
+    Server side search API V3.
```

Comment:
Are we using the same auth & semantics as the rest of API v3? If not, this will be confusing to users, and hard to document.

Reply:
Search doesn't require auth on .org, .com works with the auth backends, we could add support for `TokenAuthentication` as the other API v3 endpoints.

Not sure what you mean with semantics